sql-plugin/src/main/scala/com/nvidia/spark/rapids/higherOrderFunctions.scala

```diff
@@ -277,20 +265,148 @@ case class GpuArrayTransform(
     withResource(GpuExpressionsUtils.columnarEvalToColumn(argument, batch)) { arg =>
       val dataCol = withResource(
         makeElementProjectBatch(batch, arg.getBase)) { cb =>
         GpuExpressionsUtils.columnarEvalToColumn(function, cb)
       }
-      withResource(dataCol) { dataCol =>
-        withResource(GpuListUtils.replaceListDataColumnAsView(arg.getBase, dataCol.getBase)) {
-          retView =>
-            GpuColumnVector.from(retView.copyToColumnVector(), dataType)
+      withResource(dataCol) { _ =>
+        val cv = GpuListUtils.replaceListDataColumnAsView(arg.getBase, dataCol.getBase)
+        transformListColumnView(cv)
+      }
+    }
+  }
+}
+
+
+case class GpuArrayTransform(
+  argument: Expression,
+  function: Expression,
+  isBound: Boolean = false,
+  boundIntermediate: Seq[GpuExpression] = Seq.empty) extends GpuArrayTransformBase {
+
+  override def dataType: ArrayType = ArrayType(function.dataType, function.nullable)
+
+  override def prettyName: String = "transform"
+
+  override def bind(input: AttributeSeq): GpuExpression = {
+    val (boundFunc, boundArg, boundIntermediate) = bindLambdaFunc(input)
+
+    GpuArrayTransform(boundArg, boundFunc, isBound = true, boundIntermediate)
+  }
+
+  override protected def transformListColumnView(
+    lambdaTransformedCV: cudf.ColumnView): GpuColumnVector = {
+    withResource(lambdaTransformedCV) { cv =>
+      GpuColumnVector.from(cv.copyToColumnVector(), dataType)
+    }
+  }
+}
+
+
+case class GpuArrayExists(
+    argument: Expression,
+    function: Expression,
+    followThreeValuedLogic: Boolean,
+    isBound: Boolean = false,
+    boundIntermediate: Seq[GpuExpression] = Seq.empty) extends GpuArrayTransformBase {
+
+  override def dataType: DataType = BooleanType
+
+  override def prettyName: String = "exists"
+
+  override def nullable: Boolean = super.nullable || function.nullable
+
+  override def bind(input: AttributeSeq): GpuExpression = {
+    val (boundFunc, boundArg, boundIntermediate) = bindLambdaFunc(input)
+    GpuArrayExists(boundArg, boundFunc,  followThreeValuedLogic,isBound = true, boundIntermediate)
+  }
+
+  private def imputeFalseForEmptyArrays(
+    transformedCV: cudf.ColumnView,
+    result: cudf.ColumnView
+  ): GpuColumnVector = {
+    withResource(cudf.Scalar.fromBool(false)) { falseScalar =>
+      withResource(cudf.Scalar.fromInt(0)) { zeroScalar =>
+        withResource(transformedCV.countElements()) { elementCounts =>
+          withResource(elementCounts.equalTo(zeroScalar)) { isEmptyList =>
+            GpuColumnVector.from(isEmptyList.ifElse(falseScalar, result), dataType)
+          }
+        }
+      }
+    }
+  }
+
+  private def existsReduce(columnView: cudf.ColumnView, nullPolicy: cudf.NullPolicy) = {
+    columnView.listReduce(
+      cudf.SegmentedReductionAggregation.any(),
+      nullPolicy,
+      DType.BOOL8)
+  }
+
+  private def replaceChildNullsByFalse(cv: cudf.ColumnView): cudf.ColumnView = {
```

Comment:
nit: can we rename this to `replaceChildNullsByFalseView` or at least add in a comment to make sure that it is clear that the input cv cannot be closed until the output of this function is also closed.